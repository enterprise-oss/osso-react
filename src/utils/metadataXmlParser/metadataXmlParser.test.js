import azureMetadata from '~/__fixtures__/azure_federated_metadata.xml';
import oktaMetadata from '~/__fixtures__/okta_federated_metadata.xml';

import parseMetadataXML from './metadataXmlParser';

// For Identity Providers that support a Federated Metadata XML,
// we unit test that parsing actual files from test instances are
// properly parsed for necessary values

describe('for Okta', () => {
  test('it properly parses a federated metadata XML file', () => {
    const result = parseMetadataXML(oktaMetadata);
    expect(result).toMatchSnapshot();
  });
});

describe('for Azure', () => {
  test('it properly parses a federated metadata XML file', () => {
    const result = parseMetadataXML(azureMetadata);
    expect(result).toMatchSnapshot();
  });
});
