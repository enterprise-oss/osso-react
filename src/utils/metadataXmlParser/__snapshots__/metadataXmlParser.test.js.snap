// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`for Azure it properly parses a federated metadata XML file 1`] = `
Object {
  "ssoCert": "MIIC8DCCAdigAwIBAgIQbdTXJsn+ZbxKOYuKiFvLzTANBgkqhkiG9w0BAQsFADA0MTIwMAYDVQQDEylNaWNyb3NvZnQgQXp1cmUgRmVkZXJhdGVkIFNTTyBDZXJ0aWZpY2F0ZTAeFw0yMDAzMjgyMjQyNTBaFw0yMzAzMjgyMjQyNTBaMDQxMjAwBgNVBAMTKU1pY3Jvc29mdCBBenVyZSBGZWRlcmF0ZWQgU1NPIENlcnRpZmljYXRlMIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAoKYdA7eYHf8ZBVp7aQ9HyUlQtzPJB/0BVDabyDLJHza6OOrOenCjps04i+QO5h/AwI76ZI2WG8dvl3icm05MIBqEdlXGeiQHUHqAG0GhsGZTrtB9594Dp3NYkLjexLMs5ru+laspg8JRRn7xfeBxoxoIbwmBw1OSaq1EKntC2CygOhx4B7/i630HXIdjq3pFa45jdfXVPA/9XbqurrGQ1CMMxzec+MIiXPsQbd3EVYWpb+KSLjJAfQ8WAH75QLUiFwsNAzYWsUAjbBM+F3i7Hh0TWNGQ/wVGvUhWxrmcitT56DeeMt9fgIMpaxi8cc70mwlAWiQ+9zL+HTV4G/ORXQIDAQABMA0GCSqGSIb3DQEBCwUAA4IBAQAWwnBIGbE9BC+AqA0xDtDkfsqhWaF3Bx889Haz6EGYt2Tm8BzmpCPS9Jq2KNTPyxkMsqSQOilQT5LiIw0OBifseAd++DvYd1rVmirKpUG6p/Z1YwHRpi98+PpyOEvedUMVP+Iw66EW+98/OM2IreFpqMYSiz2Qa4PVbi/M0wZ5cccrPUbemRfM9+9912L99jbSwgfUCjo6N2LYje3fYrBtGL9t63qWeON+2mE4fBqqy9rFVeQImF1vSXOloKh8GDR14QcCPNlsUYpuezeMYQ6ppszegAs+Ahwoq7vJNrMnzlCHJfkE6Fo5NhmmJSqq/bVflUfl68bdcWkk2+JuHO8J",
  "ssoUrl": "https://login.microsoftonline.com/0af6c610-c40c-4683-9ea4-f25e509b8172/saml2",
}
`;

exports[`for Google it properly parses a federated metadata XML file 1`] = `
Object {
  "ssoCert": "MIIDdDCCAlygAwIBAgIGAXPpk/8QMA0GCSqGSIb3DQEBCwUAMHsxFDASBgNVBAoTC0dvb2dsZSBJ
bmMuMRYwFAYDVQQHEw1Nb3VudGFpbiBWaWV3MQ8wDQYDVQQDEwZHb29nbGUxGDAWBgNVBAsTD0dv
b2dsZSBGb3IgV29yazELMAkGA1UEBhMCVVMxEzARBgNVBAgTCkNhbGlmb3JuaWEwHhcNMjAwODEz
MjA0NzMzWhcNMjUwODEyMjA0NzMzWjB7MRQwEgYDVQQKEwtHb29nbGUgSW5jLjEWMBQGA1UEBxMN
TW91bnRhaW4gVmlldzEPMA0GA1UEAxMGR29vZ2xlMRgwFgYDVQQLEw9Hb29nbGUgRm9yIFdvcmsx
CzAJBgNVBAYTAlVTMRMwEQYDVQQIEwpDYWxpZm9ybmlhMIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8A
MIIBCgKCAQEAvwXqXsMZJxVhB14NJVOT4hjY80xYNqvRr1QONr0FWQRA6y9M7Auc0kS7TwYHXGNN
/hVWVfLV11pfvbK1bB3FpyZgqgjs578X3sJYpkuRvN2L5fVqCQT1RcKwB5OWIZe+pf77KCxwUlv5
nDZU0aQ0AvSXYUgjgaKCAOgCn/8MFHxqm3WRtZVuf/9TvDcmrTPRWcEXjxS45bqXbkzIR0sp/tTl
3rxYomt4m9OVrGaoXk0nh/U7bCXpw1bAPJ0+FzhBBcLXoSM5NuKWkml26E10zBBTfPL0fsrzzcg0
Ig9PHRJG71zvamFkOvt0c2eslOrj11eOTXcEc7RTGyoI8hxcxQIDAQABMA0GCSqGSIb3DQEBCwUA
A4IBAQCdzm+oRLU6YI1ABZ8FuEuSaQrLsrf7Yrg1ekLbLHSy8ls5uh7A19vsrPPevUPNfr2oKQVk
QAEWwLSyl+wvQo0hyrGt6EPrampgBihSPIy/5pcDkefgNNi82KhqtsGlojaUK7finCCuliSffoU0
FugZsL2Z80yCfaGzSy4gJZ4Th1rYjaSiW0NBTJm0Mp0HDZSenX6RY/FPpMxBxrLYGGStli+ClaHX
sKzW5egqQBYGRNxF30s4zJLySnNzvOtgBKeJyiDtYD2VHjSGSykv0yUOb5JcrnGSNTSTI00yN1eL
lhzhJ23VP0z1GC4+K1ACAGC042zTNGnrMxdtNY5orJ9k",
  "ssoUrl": "https://accounts.google.com/o/saml2/idp?idpid=C02goul2w",
}
`;

exports[`for Okta it properly parses a federated metadata XML file 1`] = `
Object {
  "ssoCert": "MIIDpDCCAoygAwIBAgIGAXEiD4LlMA0GCSqGSIb3DQEBCwUAMIGSMQswCQYDVQQGEwJVUzETMBEG A1UECAwKQ2FsaWZvcm5pYTEWMBQGA1UEBwwNU2FuIEZyYW5jaXNjbzENMAsGA1UECgwET2t0YTEU
MBIGA1UECwwLU1NPUHJvdmlkZXIxEzARBgNVBAMMCmRldi0xNjIwMjQxHDAaBgkqhkiG9w0BCQEW
DWluZm9Ab2t0YS5jb20wHhcNMjAwMzI4MTY1MTU0WhcNMzAwMzI4MTY1MjU0WjCBkjELMAkGA1UE
BhMCVVMxEzARBgNVBAgMCkNhbGlmb3JuaWExFjAUBgNVBAcMDVNhbiBGcmFuY2lzY28xDTALBgNV
BAoMBE9rdGExFDASBgNVBAsMC1NTT1Byb3ZpZGVyMRMwEQYDVQQDDApkZXYtMTYyMDI0MRwwGgYJ
KoZIhvcNAQkBFg1pbmZvQG9rdGEuY29tMIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEA
wsnP4UTfv3bxR5Jh0at51Dqjj+fKxFznzFW3XA5NbF2SlRLjeYcvj3+47TC0eP6xOsLWfnvdnx4v
dd9Ufn7jDCo5pL3JykMVEh2I0szF3RLC+a532ArcwgU9Px48+rWVwPkASS7l4NHAM4+gOBHJMQt2
AMohPT0kU41P8BEPzfwhNyiEXR66JNZIJUE8fM3Vpgnxm/VSwYzJf0NfOyfxv8JczF0zkDbpE7Tk
3Ww/PFFLoMxWzanWGJQ+blnhv6UV6H4fcfAbcwAplOdIVHjS2ghYBvYNGahuFxjia0+6csyZGrt8
H4XmR5Dr+jXY5K1b1VOA0k19/FCnHHN/smn25wIDAQABMA0GCSqGSIb3DQEBCwUAA4IBAQBgD9NE
4OCuR1+vucV8S1T6XXIL2hB7bXBAZEVHZ1aErRzktgXAMgVwG267vIkD5VOXBiTy9yNU5LK6G3k2
zewU190sL1dMfyPnoVZyn94nvwe9A+on0tmZdmk00xirKk3FJdacnZNE9Dl/afIrcNf6xAm0WsU9
kbMiRwwvjO4TAiygDQzbrRC8ZfmT3hpBa3aTUzAccrvEQcgarLk4r7UjXP7a2mCN3UIIh+snN2Ms
vXHL0r6fM3xbniz+5lleWtPFw73yySBc8znkWZ4Tn8Lh0r6o5nCRYbr2REUB7ZIfiIyBbZxIp4kv
a+habbnQDFiNVzEd8OPXHh4EqLxOPDRW",
  "ssoUrl": "https://dev-162024.okta.com/app/vcardmedev162024_someosso_1/exkato63bTfgFkFIt4x6/sso/saml",
}
`;

exports[`for OneLogin it properly parses a federated metadata XML file 1`] = `
Object {
  "ssoCert": "MIID7jCCAtagAwIBAgIUEG87INHh08jLm3W8jBuive5Fg64wDQYJKoZIhvcNAQEF
BQAwSzEWMBQGA1UECgwNRW50ZXJwcmlzZU9zczEVMBMGA1UECwwMT25lTG9naW4g
SWRQMRowGAYDVQQDDBFPbmVMb2dpbiBBY2NvdW50IDAeFw0yMDA5MTMxNjI1Mzla
Fw0yNTA5MTMxNjI1MzlaMEsxFjAUBgNVBAoMDUVudGVycHJpc2VPc3MxFTATBgNV
BAsMDE9uZUxvZ2luIElkUDEaMBgGA1UEAwwRT25lTG9naW4gQWNjb3VudCAwggEi
MA0GCSqGSIb3DQEBAQUAA4IBDwAwggEKAoIBAQDhq3fz3eH/EqlBEt5srG6asDMa
HjDYB0enovN7Z5JfaMbyOa9Ch1f7/HJ8G+fN6tGidH7+toNZGmu+F1hMrUuAEc1Z
6X7mnp8DcKneGdPKXeOrUoc8kB2Sim6Sv3dLjg8nt9/qhE4lyZGHe8qWujq3Oz2g
FOYrZ25Zm1JmzpIaBL7iqyxT9enuVuuJf/krXx59LxpUHGqnBgUiCWtV8M4q2gHZ
UFnSdyiRD9JuzgJv238hHW5d+dsWHtu6Qxxpy9m5HmU25vXTv6YCvD9JL/VRd2Or
fpCrpJxI9rL9ESUSMEk/wltp6zP/KsNyfvVzIWDX4Caxh27bQeqHl9aVFkenAgMB
AAGjgckwgcYwDAYDVR0TAQH/BAIwADAdBgNVHQ4EFgQUxQME9NAg3eDYr7lBcNdb
fMd4CvMwgYYGA1UdIwR/MH2AFMUDBPTQIN3g2K+5QXDXW3zHeArzoU+kTTBLMRYw
FAYDVQQKDA1FbnRlcnByaXNlT3NzMRUwEwYDVQQLDAxPbmVMb2dpbiBJZFAxGjAY
BgNVBAMMEU9uZUxvZ2luIEFjY291bnQgghQQbzsg0eHTyMubdbyMG6K97kWDrjAO
BgNVHQ8BAf8EBAMCB4AwDQYJKoZIhvcNAQEFBQADggEBANBD/cPNh25pr0jk8IAG
b5IoX4nVLxt2JXDjbnhtYA33nKwJMP2/BMdi+OrMuXi9NdpUXWVnUluL2BgQse4V
eO/YhkyiU6oIvC+AerO7yBuyLyvePo57uYSOOWNiWtCuKrnKEu8+zkR1D/1ppLmF
DdzPDWvpCik5vZmM0xcWvZ3x2jAFIts0auO+oa8CcecwnvYoj6w8SYVWXTt5SlkW
fJ8bsj3IxLzYjp88YrIVopj5rHnPtRa9xh+Gbg9EOGvLE8uS8M9s0ivoo2YkA18l
/ZF54PjuZ62igDiM6Jd7FNHFHF+8J0FWpb/B1pGGiTMjlGduKrNomm5kARqHEa5D
3V0=",
  "ssoUrl": "https://enterpriseoss-dev.onelogin.com/trust/saml2/http-redirect/sso/63507662-d171-4215-83ab-031f64b247ea",
}
`;

exports[`for Ping it properly parses a federated metadata XML file 1`] = `
Object {
  "ssoCert": "MIIDgDCCAmigAwIBAgIGAXW9k4k+MA0GCSqGSIb3DQEBCwUAMIGAMQswCQYDVQQG
                        EwJVUzEWMBQGA1UECgwNUGluZyBJZGVudGl0eTEWMBQGA1UECwwNUGluZyBJZGVu
                        dGl0eTFBMD8GA1UEAww4UGluZ09uZSBTU08gQ2VydGlmaWNhdGUgZm9yIEVuZCBV
                        c2VyIFNhbmRib3ggZW52aXJvbm1lbnQwHhcNMjAxMTEyMTc0OTI0WhcNMjExMTEy
                        MTc0OTI0WjCBgDELMAkGA1UEBhMCVVMxFjAUBgNVBAoMDVBpbmcgSWRlbnRpdHkx
                        FjAUBgNVBAsMDVBpbmcgSWRlbnRpdHkxQTA/BgNVBAMMOFBpbmdPbmUgU1NPIENl
                        cnRpZmljYXRlIGZvciBFbmQgVXNlciBTYW5kYm94IGVudmlyb25tZW50MIIBIjAN
                        BgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAnNOxLUf4oYV1pszJkJaFDTDDsFUp
                        lf6QiKA4k4MGgVEoiemUGxrowRhqxKrRYmXLkNv7gquq98oc6XsnnAP3Piou+GZ6
                        s/nrhPKySPOdY44iz/Wncotlqo9rvkggqDdaBDOItEEoVGCr44z0L1aMv8fs2kiK
                        yjwTVL+k1mGcOtflhyv1xrxHJIkcQcuNpy5ea4M4g2nkNW/0+eC/kaS7t9+eFXgK
                        QgZ6EK7xR0FHkj2nwsBh+LiIVkqp6pBcEcLVr/0oZoJIx7fWjTWveVDj9yyYBD6g
                        C5n09iEC+OJWNcXgSrDG3md+MnzxQ6f1Pj4Gu2Pb787a+cbJ/JthFrqy6wIDAQAB
                        MA0GCSqGSIb3DQEBCwUAA4IBAQAWHjd/O3pro/Gs2e4/MaFU+BqlsDixQwXMAk+G
                        rXlpmQObJMUJgxynBtcVZRD68jumMr5tNfXiGHnA5M5kjkCm7DurLMf+90v4BGEv
                        W2okhaAchbHhTCLZB/mJWaAKeOwFUOSuOMR0VaPvQoTWOzbFrKlWGKJaEtvC0dWo
                        DS2BpThfs4cdm05L48/vbPiBrIms/D7b1PE4aNQ4re6GC2+u+4HtNiVuq66qlNZm
                        tjW6bX+Mls1uFn6QGniBFS6JOZIQX5xXsAwglvB49b6rlg8nKGlV0an5XAqlq67N
                        azUIRICx8UqtzbaxEAocvcqJli6lP6tQpZll6uWZ99PQjqNr",
  "ssoUrl": "https://auth.pingone.com/5fe5a1f5-a0be-4d76-995b-252bc0e23852/saml20/idp/sso",
}
`;

exports[`for Salesforce it properly parses a federated metadata XML file 1`] = `
Object {
  "ssoCert": "MIIErDCCA5SgAwIBAgIOAXX/yvUfAAAAAEm5LxAwDQYJKoZIhvcNAQELBQAwgZAxKDAmBgNVBAMMH1NlbGZTaWduZWRDZXJ0XzI1Tm92MjAyMF8xNDI0NTIxGDAWBgNVBAsMDzAwRDR4MDAwMDAyMlBLTDEXMBUGA1UECgwOU2FsZXNmb3JjZS5jb20xFjAUBgNVBAcMDVNhbiBGcmFuY2lzY28xCzAJBgNVBAgMAkNBMQwwCgYDVQQGEwNVU0EwHhcNMjAxMTI1MTQyNDUyWhcNMjExMTI1MTIwMDAwWjCBkDEoMCYGA1UEAwwfU2VsZlNpZ25lZENlcnRfMjVOb3YyMDIwXzE0MjQ1MjEYMBYGA1UECwwPMDBENHgwMDAwMDIyUEtMMRcwFQYDVQQKDA5TYWxlc2ZvcmNlLmNvbTEWMBQGA1UEBwwNU2FuIEZyYW5jaXNjbzELMAkGA1UECAwCQ0ExDDAKBgNVBAYTA1VTQTCCASIwDQYJKoZIhvcNAQEBBQADggEPADCCAQoCggEBAO6izQz1ZmOoBwTuziOunwqL1UMfYaKQsIX+EPQISwVOy2Lge45O/S0ef1yy8kVKVERe9ffO6UoHeX0ETmlGgSKx6qRsJPsS7go5jMTo73N97wdWZHFUxOo3KwbOTJtxzXabPxUrv5IA4TdDFUuSG2kYPCQYuqTP1vL3qarRO3JsnpoxGJ5ti7uv7Xv4nfVl38Qn5daL/okFNyJPmmlGqIvGDvmZgk8rmPA0GtL1nqf/QuOkgHBwicXe96ga//e7tYtk/drTlTvc4LUbN2xwemrbIbzHyo/S/0c4bPy4YcqC5F/kiFPpyq0UIHEDYtqKGqWtGGAgNgYB8d1Duq1D08sCAwEAAaOCAQAwgf0wHQYDVR0OBBYEFJ1csfVxNOTDy4Tf6k6eD5bhS/KhMA8GA1UdEwEB/wQFMAMBAf8wgcoGA1UdIwSBwjCBv4AUnVyx9XE05MPLhN/qTp4PluFL8qGhgZakgZMwgZAxKDAmBgNVBAMMH1NlbGZTaWduZWRDZXJ0XzI1Tm92MjAyMF8xNDI0NTIxGDAWBgNVBAsMDzAwRDR4MDAwMDAyMlBLTDEXMBUGA1UECgwOU2FsZXNmb3JjZS5jb20xFjAUBgNVBAcMDVNhbiBGcmFuY2lzY28xCzAJBgNVBAgMAkNBMQwwCgYDVQQGEwNVU0GCDgF1/8r1HwAAAABJuS8QMA0GCSqGSIb3DQEBCwUAA4IBAQBl3kc9t/TZin63vSptAbzE5mpzs5m4rF9gr5MrH+AhA85vw0IW62Yqgmp6daQrORI93bB9udDblkl8W3oMEdGROKnfjcobZQKTNCC0Ec/6Gke+o9OLl7jevZpwTHuv200UuEqCOeexAIEMsQCJKnoHhDiXtOM63mxMD6QvprTa5ZwaFFja9IY70Mb6VkGfiZNCNttaC0jw+vpAVhtUxMx5APU9wuZ73qVshsEjmADp9y9NmGtH4/aqrU2AvYbYo9SMky4D5NDh8SCZNCE5Ihl3EBJlgjMlBKZhMAAw6C4qtpG9eN8CXSp6gHtJfin4KSuFFJAgvu7+Rr1TP+0TWiLD",
  "ssoUrl": "https://enterpriseoss-dev-ed.my.salesforce.com/idp/endpoint/HttpRedirect",
}
`;
